### DISCUSSION TITLE
How to customize the sub page to return status

### DISCUSSION THREAD

**Discussion by shijinglin** [cefsharp/CefSharp#4730] [General]
I need to customize the page to present the content of sub page .

How to return the closed status and partial data of the sub page to the main page after it is closed?

`

    internal class CefLifeSpanHandler : ILifeSpanHandler
    {
        public bool OnBeforePopup(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, string targetUrl, string targetFrameName, WindowOpenDisposition targetDisposition, bool userGesture, IPopupFeatures popupFeatures, IWindowInfo windowInfo, IBrowserSettings browserSettings, ref bool noJavascriptAccess, out IWebBrowser newBrowser)
        {
            if (!string.IsNullOrEmpty(targetUrl))
            {
                //customize 
                ChromiumWebBrowser browserTemp = chromiumWebBrowser as ChromiumWebBrowser;
                FormNew formNew = new FormNew(browserTemp, targetUrl);
                formNew.Show();
                newBrowser = null;
                return true;
            }
            //other,print...
            newBrowser = null;
            return false;
        }

        public void OnAfterCreated(IWebBrowser chromiumWebBrowser, IBrowser browser)
        {

        }

        public bool DoClose(IWebBrowser chromiumWebBrowser, IBrowser browser)
        {
            return false;
        }

        public void OnBeforeClose(IWebBrowser chromiumWebBrowser, IBrowser browser)
        {

        }
    }
`

**Comment by amaitland** [comment]
> I need to customize the page to present the content of sub page .
> How to return the closed status and partial data of the sub page to the main page after it is closed?

You are cancelling creation of the popup, so the parent->child relationship doesn't exist. It's not possible to communicate between the two using your code.

See https://github.com/cefsharp/CefSharp/wiki/General-Usage#popups
